bears/python/PythonPackageInitBear.py

```diff
@@ -0,0 +1,46 @@
+import os
+import re
+
+from coalib.results.Result import Result
+from coalib.bears.GlobalBear import GlobalBear
+from coalib.settings.ConfigurationGathering import (find_user_config)
+
+
+class PythonPackageInitBear(GlobalBear):
+    """
+    Looks for missing __init__.py files in directories containing python files.
+    """
+
+    LANGUAGES = {'Python', 'Python 3', 'Python 2'}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+
+    def run(self,
+            ignore_dirs: list = [],
+            create_empty_init_file: bool = True):
+        """
+        :param root_directory:
+            The root directory of the project whose tree has to be checked
+        :ignore_dirs:
+            List of directories to be ignored
+        """
+        files = sorted(self.file_dict)
+        dirs = set()
+        config_path = self.get_config_dir() or find_user_config(
+            os.getcwd()).strip('.coafile')
+        ignore_dirs = [os.path.join(config_path, dir) for dir in ignore_dirs]
```

Comment:
I tested if it works or not..

- My directory structure
```
.
|-- other
|   |-- dir
|   |   `-- dir.py
|   `-- other.py
`-- some
    |-- ignored
    |   |-- dir
    |   |   `-- coala.py
    |   `-- thon.py
    `-- some.py
```

- `.coafile`
```
[PythonPackageInit]
ignore = some
files = **.py
```
- output
```
➜  testDir coala -d ../coala-bears/bears/python -b PythonPackageInitBear
Executing section Default...
Executing section PythonPackageInit...

Project wide:
|    | [NORMAL] PythonPackageInitBear:
|    | Directory /home/phoenista/Desktop/meet/oss-contribution/coala/testDir/other does not contain __init__.py file

Project wide:
|    | [NORMAL] PythonPackageInitBear:
|    | Directory /home/phoenista/Desktop/meet/oss-contribution/coala/testDir/other/dir does not contain __init__.py file

Project wide:
|    | [NORMAL] PythonPackageInitBear:
|    | Directory /home/phoenista/Desktop/meet/oss-contribution/coala/testDir/some does not contain __init__.py file

Project wide:
|    | [NORMAL] PythonPackageInitBear:
|    | Directory /home/phoenista/Desktop/meet/oss-contribution/coala/testDir/some/ignored does not contain __init__.py file

Project wide:
|    | [NORMAL] PythonPackageInitBear:
|    | Directory /home/phoenista/Desktop/meet/oss-contribution/coala/testDir/some/ignored/dir does not contain __init__.py file
```

# Workaround
- `.coafile`
```
[PythonPackageInit]
ignore = some/**
files = **.py
```
- output
```
➜  testDir coala -d ../coala-bears/bears/python -b PythonPackageInitBear
Executing section Default...
Executing section PythonPackageInit...

Project wide:
|    | [NORMAL] PythonPackageInitBear:
|    | Directory /home/phoenista/Desktop/meet/oss-contribution/coala/testDir/other does not contain __init__.py file

Project wide:
|    | [NORMAL] PythonPackageInitBear:
|    | Directory /home/phoenista/Desktop/meet/oss-contribution/coala/testDir/other/dir does not contain __init__.py file
```